README.md

```diff
@@ -1,6 +1,39 @@
-# GanacheTimeTraveler
+# ganache-time-traveler
+A testing toolset that allows developers to write unit tests for the ethereum blockchain.
 
-`npm install ganache-time-traveler`
+NOTE:
+- this only works with ganache-cli
+- this only works locally
 
-example usage is here https://github.com/ejwessel/TimeContract
+## Dependencies
+- [ganache-cli](https://github.com/trufflesuite/ganache-cli)
+- [truffle](https://www.trufflesuite.com/docs/truffle/getting-started/installation)
 
+## Install
+- `npm i ganache-time-traveler`
+
+## Usage
+add `require` at the top of your tests
+```javascript
+const helper = require('ganache-time-traveler');
+```
+
+add the `beforeEach` and `afterEach` hooks into your test file
+ ```javascript
+beforeEach(async() => {
+    let snapShot = await helper.takeSnapshot();
+    snapshotId = snapShot['result'];
+});
+
+afterEach(async() => {
+    await helper.revertToSnapShot(snapshotId);
+});
+ ```
+
+## Other methods
+### advancing time
+```javascript
+helper.advanceTimeAndBlock(<seconds_to_advance_by>)
+```
+
+example is here https://github.com/ejwessel/TimeContract
```

Comment:
maybe we should move this example into this repo, but exclude it from the npm publish

Reply:
Yeah i was thinking about having a separate examples folder, with some usages of the library.

Reply:
lets do it
make a new PR for it.